service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -65,4 +75,63 @@ protected static boolean isLastPage(JSONObject response) throws JSONException {
         }
         return result;
     }
+
+    public StatisticalIndicatorDataDimension getTimeperiodDimensionFromIndicatorData(String timePeriodDimensionId,
+            String indicator) {
+        UnsdRequest request = new UnsdRequest(config);
+        request.setIndicator(indicator);
+
+        StatisticalIndicatorDataDimension selector = new StatisticalIndicatorDataDimension(timePeriodDimensionId);
+        Map<Integer, Integer> countOfAreaCodesForYear = new HashMap<>();
+
+        try {
+            while (true) {
+                JSONObject response = JSONHelper.createJSONObject(request.getIndicatorData(null));
+                parseTimePeriod(countOfAreaCodesForYear, response);
+                if (isLastPage(response)) {
+                    break;
+                }
+                request.nextPage();
+            }
+
+            List<IdNamePair> allowedValues = getSortedListOfYearsThatBelongToSeveralGeoAreas(
+                    countOfAreaCodesForYear);
+
+            selector.setAllowedValues(allowedValues);
+            return selector;
+
+        } catch (JSONException ex) {
+            throw new APIException("Error during parsing UNSD data response.", ex);
+        }
+    }
+
+    public static void parseTimePeriod(Map<Integer, Integer> countOfAreaCodesForYear, JSONObject response) {
+        try {
+            JSONArray data = response.getJSONArray(DATA_KEY);
+            for (int i = 0; i < data.length(); i++) {
+                JSONObject o = (JSONObject) data.get(i);
+                Integer year = getTimeperiodYear(o);
+
+                if (countOfAreaCodesForYear.containsKey(year)) {
```

Comment:
This if-else is very readable and I'm not asking you to change it, but for fun and curiosity I'll note that `countOfAreaCodesForYear.merge(year, 1, Integer::sum);` would've done the same thing.

Reply:
Changed code as proposed.